### PR TITLE
[5.0][Events] Privacy event classes

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

Implementing Event classes for Privacy events.

Affected events:

- [x] onPrivacyCollectAdminCapabilities
- [x] onPrivacyCheckPrivacyPolicyPublished
- [x] onPrivacyExportRequest
- [x] onPrivacyCanRemoveData
- [x] onPrivacyRemoveData

### Testing Instructions

Use Privacy features, export request, delete reques, capabilities overview

### Actual result BEFORE applying this Pull Request

Works

### Expected result AFTER applying this Pull Request

Works

### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: https://docs.joomla.org/Plugin/Events#Privacy
- [ ] No documentation changes for docs.joomla.org needed
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/177
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
please resolve conflict

File: administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php [hunk 1]
```diff
@@ -46,10 +49,15 @@ public static function getPrivacyPolicyInfo()
          * Prior to 3.9.0 it was common for a plugin such as the User - Profile plugin to define a privacy policy or
          * terms of service article, therefore we will also import the user plugin group to process this event.
          */
-        PluginHelper::importPlugin('privacy');
-        PluginHelper::importPlugin('user');
-
-        Factory::getApplication()->triggerEvent('onPrivacyCheckPrivacyPolicyPublished', [&$policy]);
+        PluginHelper::importPlugin('privacy', null, true, $dispatcher);
+        PluginHelper::importPlugin('user', null, true, $dispatcher);
```
Comment (HLeithner) [inline administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php:53]: for later, have we a chance to deprecate this? if or if not can you add this to the migration documentation to not use "user" plugins for privacy stuff so we can remove this import here in Joomla 7.0

**Comment by HLeithner** [comment]
Merging this as with code review and based on the other similar PRs, before beta1 to get enough feedback for final release.

File: administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php (hunk 1 above)
Comment (Fedik) [reply administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php:53]: I not sure what a reason for these imports, But I think we can just add deprecation here. It is our module.
And 'user' group should be already imported, while login.

Comment (HLeithner) [reply administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php:53]: it has a description 2 lines above that's it's b/c for < 3.9 ;-)

I'm not sure if we always import `user` plugins for each page load?

Comment (Fedik) [reply administrator/modules/mod_privacy_status/src/Helper/PrivacyStatusHelper.php:53]: hmhm, probably not, I thought it always up with onUserAuthorisation,
but not sure currently

**Pull request merged**